otherlibs/systhreads/st_posix.h

```diff
@@ -356,19 +356,17 @@ static void st_decode_sigset(value vset, sigset_t * set)
 
 static value st_encode_sigset(sigset_t * set)
 {
-  value res = Val_int(0);
+  CAMLparam0();
+  CAMLlocal1(res);
   int i;
 
-  Begin_root(res)
-    for (i = 1; i < NSIG; i++)
-      if (sigismember(set, i) > 0) {
-        value newcons = caml_alloc_small(2, 0);
-        Field(newcons, 0) = Val_int(caml_rev_convert_signal_number(i));
-        Field(newcons, 1) = res;
-        res = newcons;
-      }
-  End_roots();
-  return res;
+  res = Val_emptylist;
+
+  for (i = 1; i < NSIG; i++)
+    if (sigismember(set, i) > 0) {
+      res = caml_alloc_2(2, Val_int(caml_rev_convert_signal_number(i)), res);
```

Comment:
I'm not to sure about this, but shouldn't the first parameter `2` here be `Tag_cons` (`0`) (the tag of the first cell) instead?

Reply:
Yes, you are right.